mkosi

```diff
@@ -2830,25 +2865,46 @@ def check_root():
     if os.getuid() != 0:
         die("Must be invoked as root.")
 
+def run_shell(args):
+    target = "--directory=" + args.output if args.output_format in (OutputFormat.directory, OutputFormat.subvolume) else "--image=" + args.output
+
+    cmdline = ["systemd-nspawn",
+               target]
+
+    if args.verb == "boot":
+        cmdline += ('--boot',)
```

Comment:
Yeah, we should add that eventually. I didn't want to do that for now, since doing arbitrary commands should really imply --as-pid2, while invoking an interactive shell shouldn't (see my comments above), and I didn't want to get into that mess... UNIX semantics on this aren't pretty, and we already failed to do this right on nspawn itself, as --as-pid2 is not the implied default for non-interactive stuff...